go/build-and-test/action.yml

```
@@ -0,0 +1,150 @@
+name: "Go Build & Test"
+description: "Build and test go repo"
+
+inputs:
+  FLIPPCIRCLECIPULLER_REPO_TOKEN:
+    description: Flipp circleci repo token
+    required: true
+  BUF_BUILD_USER:
+    description: Buf CI user stored as secret
+    required: false
+  BUF_BUILD_API_TOKEN:
+    description: Buf API token stored as secret
+    required: false
+  SLACK_CHANNEL_ID:
+    description: The Slack channel ID(s) to send the data to
+    required: false
+  SLACK_BOT_TOKEN:
+    description: The Slack bot token to pass the data to
+    required: false
+  DB_HOST:
+    description: Database host
+    required: false
+  DB_USER:
+    description: Database user
+    required: false
+  DB_PASSWORD:
+    description: Database password
+    required: false
+  DB_PORT:
+    description: Database port
+    required: false
+  TEST_TAGS:
+    description: Test tags
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set branch variable
+      id: set-branch
+      shell: bash
+      run: |
+        if [ "${{ github.event_name }}" == "pull_request" ]; then
+          branch="${{ github.event.pull_request.head.ref }}"
+        else
+          branch="${{ github.ref }}"
+          branch="${branch#refs/heads/}"
+        fi
+        echo "branch=${branch}" >> $GITHUB_OUTPUT
+    - name: Get short sha
+      id: get-short-sha
+      run: echo "short_sha=`echo ${GITHUB_SHA::7}`" > $GITHUB_OUTPUT
+      shell: bash
+    - name: Configure Go environment
+      uses: wishabi/github-actions/go/configure@v0
+      with:
+        FLIPPCIRCLECIPULLER_REPO_TOKEN: ${{ inputs.FLIPPCIRCLECIPULLER_REPO_TOKEN }}
+        BUF_BUILD_USER: ${{ inputs.BUF_BUILD_USER }}
+        BUF_BUILD_API_TOKEN: ${{ inputs.BUF_BUILD_API_TOKEN }}
+    - name: Generate local protos
+      run: make generate-buf
```

this would break for any non-pda repo that doesn't have a make file with this command.
Would it make sense to move these team specific actions into a team folder to avoid confusion?

---

Good call, could we just do this?
```suggestion
      run: |
        buf generate
	go generate ./...
```

---

Looks like this worked! https://github.com/wishabi/distribution-api/actions/runs/6971579247/job/18981313019